src/main/resources/default/assets/javascript/client-pagination.js

```diff
@@ -0,0 +1,52 @@
+/*
+ * Made with all the love in the world
+ * by scireum in Remshalden, Germany
+ *
+ * Copyright by scireum GmbH
+ * http://www.scireum.de - info@scireum.de
+ */
+
+function paginate(paginationControls, update) {
+    const pagination = {
+        previousBtn: $("<li><a href='#' aria-label='Previous'><span aria-hidden='true'>&laquo;</span></a></li>"),
+        pageIndicator: $("<li><a href='#' aria-label='Page'/></li>"),
+        nextBtn: $("<li><a href='#' aria-label='Next'><span aria-hidden='true'>&raquo;</span></a></li>"),
+        currentPage: 1
+    }
+
+    pagination.reset = function () {
```

Comment:
this is not really proper "object oriented" JS code. My suggestion would be to extract it into a "constructor" function, append the methods to the prototype of the new type and then create a new instance of this "class" in the paginate function.
Inspiration could be taken from here: https://github.com/scireum/sirius-biz/blob/develop/src/main/resources/default/assets/javascript/multiLanguageField.js#L9

Would remove the need of always creating the structure from scratch when using the paginate function multiple times.